### ISSUE TITLE
Возможность сбросить кеш, или автоматическая инвалидация по хешам/file modified time

### ISSUE THREAD

**Issue by fe3dback** [SerafimArts/Properties#11]
Привет, ~~похоже что не работает, когда класс имлементирует какой-либо интерфейс~~. Проблема была с кешом. Ниже описал в чём дело.

Мб стоит добавить метод для удаления кеша, или автоматическую инвалидацию?

**Comment by fe3dback** [comment]
Еще потестировал, похоже проблема связана с кешем. 

Воспроизводится так:
Сначала нужно установить свойство как
```php
 @property string name
```

затем запустить код, проверка пройдет, значение обновится, кеш зафиксирует всё.

Если поменять на
```php
 @property-read
```

- то будет работать как будто там просто @property (что логично, так как это из кеша грузится). 

Но поведение не очень ожидаемое. Я думал что там будет сравнение modified_time файлов, или контрольные суммы с автоматической инвалидацией кеша.

Также в коде вроде как нет возможности руками сбросить кеш, единственный способ удалить весь кеш приложения, или хранить кеш библиотеки с другим драйвером (обычно psr16 не настраивается для множественного хранения).

**Comment by SerafimArts** [comment]
Хм, обычно персистентный кеш запускается только в продакшене, а при деплое он сбрасывается в ноль. Именно этим я руководствовался, когда делали харкорное кеширование без проверок актуальности.

Но чисто теоретически можно привязать к какому-нибудь хешу, да. Только я боюсь что объём кеша будет тогда только увеличиваться, если его не чистить постоянно.

**Comment by fe3dback** [comment]
Да, так и есть. Но иногда не выгодно при деплои сбрасывать весь кеш. Хочется иметь возможность контролировать этот аспект, как например в twig, доктрине и т.п. Я к примеру использую symfony/console, в ней создаю команду dropCache, а внутри вызов всех сбросов кешей из всех библиотек. Команда вызывается при деплое, и не удаляет пользовательский кеш данных.

Насчёт проверок, в проде точно не нужно. А на деве можно использовать стратегию кеш + проверки, если разница во времени выполнения между "без кеша" и "с кешом и проверкой" большая.
